### ISSUE TITLE
docker does not build with systemd 230 (unstable branch)

### ISSUE THREAD

**Issue by anderspapitto** [NixOS/nixpkgs#15796]
### Steps to reproduce

nix-shell -p docker

## Technical details

[anders@gurney:~]
$ nixos-version 
16.09.git.ef7e753 (Flounder)
[anders@gurney:~]
$ nix-env --version
nix-env (Nix) 1.11.2
[anders@gurney:~]
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"16.09.git.ef7e753"

**Comment by anderspapitto** [comment]
error message is

```
...
---> Making bundle: dynbinary (in bundles/1.10.3/dynbinary)
Created binary: bundles/1.10.3/dynbinary/dockerinit-1.10.3
Building: bundles/1.10.3/dynbinary/docker-1.10.3
# pkg-config --cflags libsystemd-journal
Package libsystemd-journal was not found in the pkg-config search path.
Perhaps you should add the directory containing `libsystemd-journal.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libsystemd-journal' found
pkg-config: exit status 1
builder for ‘/nix/store/75icrrwznljd766zn18h172hsdmdfv66-docker-1.10.3.drv’ failed with exit code 1
error: build of ‘/nix/store/75icrrwznljd766zn18h172hsdmdfv66-docker-1.10.3.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

**Comment by anderspapitto** [comment]
I'm guessing it's to do with the new systemd because I saw a reference to .pc file treatment changing somewhere; but that's actually just a guess

**Comment by anderspapitto** [comment]
changing `DOCKER_BUILDTAGS = [ "journald" ]` to `DOCKER_BUILDTAGS = [  ]
`in pkgs/applications/virtualization/docker fixes the build; i'm not sure if it causes runtime issues.

**Comment by bjornfor** [comment]
Hm, why am I unable to reproduce this? I have nix.useChroot set and doing "nix-build -A docker" and "nix-shell -p docker" works just fine.

**Comment by bjornfor** [comment]
Also, ```nix-shell -p docker -I nixpkgs=.``` works in the very latest nixpkgs master branch (e7c96842175da58a6cedb75c68ebe7dfbb7a342d).

**Comment by vcunat** [comment]
Seems fixed by f0de1c75b5.
